samplers/samplers_test.go

```diff
@@ -255,6 +257,57 @@ func TestSetMergeMetric(t *testing.T) {
 	assert.True(t, -1 <= countDifference && countDifference <= 1, "counts did not match after merging (%d and %d)", count1, count2)
 }
 
+func digest(values []float64) *tdigest.MergingDigest {
+	td := tdigest.NewMerging(100, false)
+	for _, v := range values {
+		td.Add(v, 1.0)
+	}
+	return td
+}
+
+// ensure histogram with no local samples flush aggregates for global flushes
+// but no aggregates for non-global (mixed scope) flushes.
+func TestGlobalHistoFlushBehavior(t *testing.T) {
```

Comment:
Instead of writing new E2E tests, I wrote these unit tests to validate this behavior change. The other behavior changes (forwarding, mainly) were covered by adding new assertions.